compiler/release

```diff
@@ -12,87 +12,138 @@
 # See README.adoc.
 # ----------------------------------------------------------------------
 
+# Print and evaluate a command
+evalp()
+{
+  echo "$@"
+  $@
+}
+
+# Get tool names from a directory
+get_tool_names()
+{
+  dir=$1
+  for file in $dir/*.jar
+  do
+    basename $file .jar
+  done
+}
+
+# Print a phase of the process
+print_phase()
+{
+  echo "----------------------------------------------------------------------"
+  echo $1
+  echo "----------------------------------------------------------------------"
+}
+
+# Check that GRAALVM_JAVA_HOME is set
 if test -z "$GRAALVM_JAVA_HOME"
 then
-  echo "release: environment variable GRAALVM_JAVA_HOME is not set" 1>&2
+  echo "[ERROR] Environment variable GRAALVM_JAVA_HOME is not set" 1>&2
   exit 1
 fi
-directory=`dirname $0`
-native_bin="$directory/native-fpp-`uname`-`uname -m`"
+
+# Run in the directory where this script is located
+cd `dirname $0`
+
+# Set local variables
+native_bin="native-fpp-`uname`-`uname -m`"
 native_image="$GRAALVM_JAVA_HOME/bin/native-image"
-meta_inf_dir="$directory/lib/src/main/resources/META-INF/native-image/"
+meta_inf_dir="lib/src/main/resources/META-INF/native-image"
+release_tgz="$native_bin.tar.gz"
+# Use gtar if it's available
+# The default tar has issues on Mac OS on GitHub
+if which gtar
+then
+  tar=`which gtar`
+else
+  tar=tar
+fi
 
 # Make directories
 mkdir -p "$meta_inf_dir"
 mkdir -p "$native_bin"
 
-# Versioning information
+print_phase "Version information"
+
+# Print version information
 echo "C compiler version"
 cc --version
-echo "Java version"
-java --version
 echo "Native Image Version"
-native-image --version
+$native_image --version
 
+print_phase "Installing JVM tools in bin"
 
-# Install jar files here
-$directory/install-trace
-# Test a run before running test
-$directory/bin/fpp-locate-defs tools/fpp-locate-defs/test/defs/defs-1.fpp tools/fpp-locate-defs/test/defs/defs-2.fpp 1>/dev/null
-if [ $? -ne 0 ]
-then
-    echo "[ERROR] Failed to run JAR with tracing enabled"
-    exit 1
-fi
+# Install jar files in bin
+./install
```

Comment:
Don't we need to run the trace of the tools?

Reply:
The current plan is to run the trace manually and correct any issues before submitting the release to GitHub. The trace results are checked in, so running the trace on GitHub should not be necessary. We run the unit tests in on GitHub on the native binaries, so if anything goes wrong the release process will catch it.